### PR TITLE
aws_byte_cursor_starts_with(input, prefix)

### PR DESCRIPTION
**Pull request by graebm**
more string helper functions

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: source/byte_buf.c [hunk 1]
```diff
@@ -558,6 +558,42 @@ bool aws_byte_cursor_eq_c_str_ignore_case(const struct aws_byte_cursor *const cu
...
+
+    if (input->len < prefix->len) {
+        return false;
+    }
+
+    struct aws_byte_cursor start = {.ptr = input->ptr, .len = prefix->len};
+    bool rv = aws_byte_cursor_eq(&start, prefix);
+
+    AWS_POSTCONDITION(aws_byte_cursor_is_valid(input));
+    AWS_POSTCONDITION(aws_byte_cursor_is_valid(prefix));
```
Comment (TingDaoK) [inline source/byte_buf.c:574]: trivial/question: they are both const, do we still need to check they are valid afterward?

**Review by TingDaoK — APPROVED** [review]
Question and 🚢

File: source/byte_buf.c (hunk 1 above)
Comment (TingDaoK) [reply source/byte_buf.c:574]: ignore it. you can still change the content inside...

Comment (graebm) [reply source/byte_buf.c:574]: I was just doing checks the same way that other const functions in this file were doing it 🤷‍♀️

**Pull request merged**